openupgradelib/openupgrade.py

```diff
@@ -2606,19 +2608,88 @@ def delete_record_translations(cr, module, xml_ids):
-        logged_query(
-            cr,
-            query,
-            (
-                module,
-                row[0] + ",%",
-                row[1],
-            ),
-        )
+        model = row[0]
+        record_id = row[1]
+        if version_info[0] < 16:
+            if not field_list:
+                query = """
+                    DELETE FROM ir_translation
+                    WHERE module = %s AND name LIKE %s AND res_id = %s;
+                """
+                logged_query(
+                    cr,
+                    query,
+                    (
+                        module,
+                        model + ",%",
+                        record_id,
+                    ),
+                )
+            else:
+                name_comparisons = " OR ".join(
+                    ["name = %s" % (model + "," + field) for field in field_list]
+                )
+                query = """
+                    DELETE FROM ir_translation
+                    WHERE module = %s AND ({name_comparison}) AND res_id = %s;
+                """.format(
+                    name_comparison=name_comparisons
+                )
+                logged_query(
+                    cr,
+                    query,
+                    (
+                        module,
+                        record_id,
+                    ),
+                )
+        else:
+            from psycopg2.extras import Json
+
+            table = model.replace(".", "_")
+            # we use information_schema to assure the columns exist
+            cr.execute(
+                """
+                SELECT isc.column_name
+                FROM information_schema.columns isc
+                JOIN ir_model_fields imf ON (
+                    imf.name = isc.column_name AND imf.model = %s)
+                WHERE isc.table_name = %s AND imf.translate""",
+                (model, table),
+            )
+            list_columns = [x[0] for x in cr.fetchall()]
+            if field_list:
+                list_columns = [x for x in list_columns if x in field_list]
+            if not list_columns:
+                continue
+            columns = ", ".join(list_columns)
+            query = """
+                SELECT {columns}
+                FROM {table}
+                WHERE id = %s""".format(
+                table=table,
+                columns=columns,
+            )
+            cr.execute(query, (record_id,))
+            list_translations = cr.fetchall()[0]
+            new_values = []
+            for column, translations in zip(list_columns, list_translations):
+                field_translations = translations or {}
+                if "en_US" in field_translations:
+                    clean_value = {"en_US": field_translations["en_US"]}
+                    new_values += [Json(clean_value)]
+                else:
+                    new_values += [None]
+            query = "UPDATE {table} SET ({columns}) = ({values}) WHERE id = %s".format(
```

Comment:
Can we easily avoid to update if no change?

Reply:
How?

Reply:
Done